core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,306 @@
+mod chunking;
+
+use anyhow::Context as _;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use prometheus_exporter::PrometheusExporterConfig;
+use std::cmp::max;
+use std::future::Future;
+use std::pin::Pin;
+use std::time::Duration;
+use tokio::sync::watch;
+use tokio::sync::watch::Receiver;
+use vise::Unit;
+use vise::{Buckets, Gauge, Histogram, Metrics};
+use zksync_config::configs::PrometheusConfig;
+use zksync_config::{PostgresConfig, SnapshotsCreatorConfig};
+
+use crate::chunking::get_chunk_hashed_keys_range;
+use zksync_dal::ConnectionPool;
+use zksync_env_config::object_store::SnapshotsObjectStoreConfig;
+use zksync_env_config::FromEnv;
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::snapshots::{
+    SnapshotFactoryDependencies, SnapshotStorageLogsChunk, SnapshotStorageLogsStorageKey,
+};
+use zksync_types::zkevm_test_harness::zk_evm::zkevm_opcode_defs::decoding::AllowedPcOrImm;
```

Comment:
??? That's a weird way to convert `u32` to `u64` 🤔 You should use `Into::into()` instead.

Reply:
yeaaah, I'm not sure what I did here, I think my IDE suggested it and I did not see where it comes from :D